pkg/cosigner/image_signer.go

```diff
@@ -0,0 +1,91 @@
+package cosigner
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/BurntSushi/toml"
+	"github.com/buildpacks/lifecycle"
+	"github.com/sigstore/cosign/cmd/cosign/cli"
+)
+
+type ImageSigner struct {
+	Logger *log.Logger
+}
+
+var (
+	cliSignCmd     = cli.SignCmd
+	secretLocation = "/var/build-secrets"
+)
+
+func NewImageSigner(logger *log.Logger) *ImageSigner {
+	return &ImageSigner{
+		Logger: logger,
+	}
+}
+
+func (s *ImageSigner) Sign(reportFilePath string, annotations map[string]interface{}) error {
+	var report lifecycle.ExportReport
+	_, err := toml.DecodeFile(reportFilePath, &report)
```

Comment:
Both Notary and Cosign are unmarshalling the report. Can we extract that to completion and pass in the report here?